volatility3/framework/plugins/windows/handles.py

```diff
@@ -54,13 +54,14 @@ def _decode_pointer(self, value, magic):
         This function mimics the decoding routine so we can generate the
         proper pointer values as well.
         """
+        kernel = self.context.modules[self.config['kernel']]
+        virtual = kernel.layer_name
 
+        # TODO: Make this logic consistent with how the windows kernel does this.
         value = value & 0xFFFFFFFFFFFFFFF8
-        value = value >> magic
-        # if (value & (1 << 47)):
-        #    value = value | 0xFFFF000000000000
+        value = value >> magic        
 
-        return value
+        return self.context.layers[virtual].get_canonical_address(value)
```

Comment:
As best I can see, this is the only real change in the request, and it's for a plugin, which can be changed in the plugin.  Essentially, plugins must beware when passing offsets, which is exactly how things are at the moment.

Reply:
Actually the problem is that the the plugin _assumes_ that canonical addresses are valid which they are **not**.